parsing/parchmint-parser.js

```diff
@@ -116,46 +126,87 @@ class ParchmintParser {
         this.valid = true;
         this.idSet = new Set();
 
+        this.layers = [];
         this.components = new Map();
         this.connections = new Map();
         this.compFeatures = new Map();
         this.connFeatures = new Map();
     }
 
+
     /**
-     * Initialize the name and id fields of a ParchKey object from a JSON
-     * object.
+     * Parse a JSON object for the layers key.
      *
-     * This method is intended for use on any subclass of ParchKey.
+     * Fills the layers array with Layer objects. If Components or Connections
+     * exist in their respective maps, under Layer IDs that do not exist in the
+     * "layers" key in the Parchmint, then the parser is set invalid.
      *
      * @since 1.0.0
      *
-     * @param {object}  parchKeyObj The ParchKey object to initialize.
-     * @param {object}  jsonObj     The JSON object with which to initialize.
+     * @param {object} jsonObj  A parsed JSON object representing the Parchmint
+     *                          file.
      */
-    initParchKey(parchKeyObj, jsonObj) {
-        parchKeyObj.name = jsonObj['name'];
-        parchKeyObj.id = jsonObj['id'];
+    parseLayers(jsonObj) {
+        // First make sure that neither components nor connections has extra/invalid layers
+        for (let layerID of this.components.keys()) {
+            let found = false;
+            for (let l of jsonObj['layers']) {
+                if (l.id === layerID) {
+                    found = true;
+                    break;
```

Comment:
Originally I wanted to use something comparable to a list comprehension in Python to get an array of IDs, but [according to Mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Array_comprehensions) they're not standard, so I decided to go with this mess.

Reply:
Can you not use the [Array includes function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)? It looks like its supported in everything except IE (which I'm more than happy not to support)

Reply:
```jsonObj['layers']``` returns an array of objects with the fields ```id``` and ```name```. In order to use the ```includes``` function, it would have to be an array of ```id```s. I tried [```Array.map```](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) and [```Array.filter```](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), but neither (as far as I am aware) allow the access of specific fields from an object.

Reply:
From [this StackOverflow article](https://stackoverflow.com/questions/19590865/from-an-array-of-objects-extract-value-of-a-property-as-array) can you do something like this?

```js
if (jsonObj['layers'].map(j => j.id).include(layerId)) { ... }
```

Reply:
That is exactly what I wanted to do. Those two loops have been changed.